### PR TITLE
Signup form

### PR DESCRIPTION
**Pull request by he-is-james**
uses next-input-field
incorporates Config but unsure that it works since it didn't work on my system
otherwise, pressing return on finishing a field moves to the next and pressing the submit button creates a user on the airtable form

### CODE REVIEW

File: android/app/src/main/java/com/apifm/MainApplication.java [hunk 1]
```diff
@@ -4,6 +4,8 @@
 import android.content.Context;
 import com.facebook.react.PackageList;
 import com.facebook.react.ReactApplication;
+import com.lugg.ReactNativeConfig.ReactNativeConfigPackage;
```
Comment (hwang635) [inline android/app/src/main/java/com/apifm/MainApplication.java:7]: can you remove these changes / this file and also the gradlew file below

File: android/settings.gradle [hunk 2]
```diff
@@ -1,4 +1,8 @@
 rootProject.name = 'APIFM'
+include ':react-native-config'
+project(':react-native-config').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-config/android')
+include ':react-native-config'
```
Comment (hwang635) [inline android/settings.gradle]: duplicate lines here

File: src/SignIn/SignInScreen.js [hunk 3]
```diff
@@ -26,6 +39,55 @@ const styles = StyleSheet.create({
...
+    ], (err, records) => {
+      if (err) {
+        console.error(err);
+        return;
+      }
+      records.forEach((record) => {
+        console.log(record.getId());
+      });
+    });
+    // console.log(firstName);
```
Comment (hwang635) [inline src/SignIn/SignInScreen.js]: remove these comments + other commented out console.log statements below

File: src/SignIn/SignInScreen.js [hunk 4]
```diff
@@ -26,6 +39,55 @@ const styles = StyleSheet.create({
...
+          password,
+          'first name': firstName,
+          'last name': lastName,
+          'phone number': number,
+          address,
+        },
+      },
+    ], (err, records) => {
+      if (err) {
+        console.error(err);
```
Comment (hwang635) [inline src/SignIn/SignInScreen.js]: Can you replace the console.log statements with an alert/etc that shows up on screen?

File: src/SignIn/SignInScreen.js [hunk 5]
```diff
@@ -26,6 +39,55 @@ const styles = StyleSheet.create({
...
+        return;
+      }
+      records.forEach((record) => {
+        console.log(record.getId());
```
Comment (hwang635) [inline src/SignIn/SignInScreen.js]: Same as above, we don't want to keep console.log statements bc they can accumulate and be confusing for devs if you don't know where they originate and they don't give any error info to the user

File: src/SignIn/SignInScreen.js [hunk 6]
```diff
@@ -40,6 +102,77 @@ export default function SignInScreen({ navigation }) {
...
+          value={password}
+          onChangeText={setPassword}
+          placeholder="password"
+          returnKeyType="next"
+          secureTextEntry={!!hidePass}
+          onSubmitEditing={() => { numberInput.current.focus(); }}
+          blurOnSubmit={false}
+          ref={passwordInput}
+        />
+        <Icon
```
Comment (hwang635) [inline src/SignIn/SignInScreen.js:155]: The hide password fx is super cool

**Review by hwang635 — CHANGES_REQUESTED** [review]
This looks really good! Please get react native config fixed if you haven't (I added steps in the env setup doc) and there are some minor edits to make that I commented on, but overall you did great work and everything looks super nice. Thanks James and Grace for your attention to detail and getting it done so fast!

**Review by hwang635 — CHANGES_REQUESTED** [review]
This looks great! One last thing -- can you add a check so that the sign up button can't be pressed/the form can't submitted unless the user has filled out all information? An alert that the user can't submit because of missing info would be great also so that they're not confused why it's not working. Right now the form can be submitted even if there's no information entered, which makes it possible to fill the airtable up with random empty rows or have a user with missing data.

**Comment by glm1550** [comment]
the form now checks if fields are populated for valid submission

File: src/SignIn/SignInScreen.js [hunk 7]
```diff
@@ -41,46 +41,80 @@ const styles = StyleSheet.create({
...
-          'last name': lastName,
-          'phone number': number,
-          address,
+    if (firstName == ""){
+      Alert.alert("Please enter the First Name to proceed")
+    }
+    else if (lastName == ""){
+      Alert.alert("Please enter the Last Name to proceed")
+    }
+    else if (email== ""){
```
Comment (hwang635) [inline src/SignIn/SignInScreen.js]: The == should be "==="

File: src/SignIn/SignInScreen.js [hunk 8]
```diff
@@ -41,46 +41,80 @@ const styles = StyleSheet.create({
...
+      setBusPhone('');
+      setBusPhone('');
+      setAddress('');
+    }
   };
 
   return (
     <View style={styles.container}>
-      <Title style={styles.titleText}> SignInScreen  :) </Title>
+      {/* <Title style={styles.titleText}> SignInScreen  :) </Title>
```
Comment (hwang635) [inline src/SignIn/SignInScreen.js]: Uncomment this part or remove it

File: src/SignIn/SignInScreen.js [hunk 9]
```diff
@@ -41,46 +41,80 @@ const styles = StyleSheet.create({
...
-    base('Users').create([
-      {
-        fields: {
-          email,
-          password,
-          'first name': firstName,
-          'last name': lastName,
-          'phone number': number,
-          address,
+    if (firstName == ""){
```
Comment (hwang635) [inline src/SignIn/SignInScreen.js]: This looks good but I don't think it passes eslint, make sure estlint is working in your env and don't push before checking that everything passes please!

File: src/SignIn/SignInScreen.js [hunk 10]
```diff
@@ -41,46 +41,80 @@ const styles = StyleSheet.create({
...
-    });
-    setFirstName('');
-    setLastName('');
-    setEmail('');
-    setPassword('');
-    setNumber('');
-    setAddress('');
+      ], (err) => {
+        if (err) {
+          alert(err);
```
Comment (hwang635) [inline src/SignIn/SignInScreen.js]: Can this be an Alert.alert like the ones above?

**Review by hwang635 — CHANGES_REQUESTED** [review]
Hi Grace and James! Thanks for adding a submission check, everything looks pretty good. There are some minor edits to make and please pull from master to resolve merge conflicts; then it should be good to merge.

**Review by hwang635 — APPROVED** [review]
Beautiful thanks for working on this!

**Pull request merged**